core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -172,17 +215,31 @@ class SpendReportingService(userInfo: UserInfo, dataSource: SlickDataSource, big
       .setArrayType(StandardSQLTypeName.STRING)
       .setArrayValues(queryParameterArrayValues)
       .build()
-}
+  }
+
+  private def getQuery(aggregationKeys: Set[SpendReportingAggregationKey], tableName: String): String = {
+    s"""
+       | SELECT
+       |  SUM(cost) as cost,
+       |  SUM(IFNULL((SELECT SUM(c.amount) FROM UNNEST(credits) c), 0)) as credits,
+       |  currency ${aggregationKeys.map(_.bigQueryAliasClause()).mkString}
+       | FROM `$tableName`
+       | WHERE billing_account_id = @billingAccountId
+       | AND _PARTITIONTIME BETWEEN @startDate AND @endDate
+       | AND project.id in UNNEST(@projects)
+       | GROUP BY currency ${aggregationKeys.map(_.bigQueryGroupByClause()).mkString}
+       |""".stripMargin
+  }
 
-  def getSpendForBillingProject(billingProjectName: RawlsBillingProjectName, startDate: DateTime, endDate: DateTime, aggregationKey: Option[SpendReportingAggregationKey] = None): Future[SpendReportingResults] = {
+  def getSpendForBillingProject(billingProjectName: RawlsBillingProjectName, startDate: DateTime, endDate: DateTime, aggregationKeyParameters: Set[SpendReportingAggregationKeyWithSub] = Set.empty): Future[SpendReportingResults] = {
     validateReportParameters(startDate, endDate)
     requireAlphaUser() {
       requireProjectAction(billingProjectName, SamBillingProjectActions.readSpendReport) {
         for {
           spendExportConf <- getSpendExportConfiguration(billingProjectName)
           workspaceProjectsToNames <- getWorkspaceGoogleProjects(billingProjectName)
-
-          query = getQuery(aggregationKey, spendExportConf.spendExportTable.getOrElse(spendReportingServiceConfig.defaultTableName))
+          aggregationKeys = aggregationKeyParameters.flatMap(maybeKeys => Set(Option(maybeKeys.key), maybeKeys.subAggregationKey).flatten)
```

Comment:
Nit: what does `maybeKeys` mean?

Reply:
ah this could be clearer, I'll add an explanatory comment.

This line unboxes the `aggregationKeyParameters` (all of which are of type `SpendReportingAggregationKeyWithSub(key: SpendReportingAggregationKey, subKey: Option[SpendReportingAggregationKey])`) and makes a `Set` of all the keys. It's a way of handling the possible `None` values and generating a set of keys that can be easily handled by `getQuery`